### DISCUSSION TITLE
SD Card Write Cycles

### DISCUSSION THREAD

**Discussion by dudududodododedede** [gtxaspec/wz_mini_hacks#77] [General]
I may be worrying about nothing but just want confirmation for peace-of-mind: is there any long-term risks to our SD Cards due to increased write cycles posed by wz_mini_hacks? In particular the concern would be with frequent random small writes i.e. log files, temp files

**Comment by gtxaspec** [comment]
good question... I think the writes done by the camera for the video, photo stills, motion recordings, etc would be the greatest load on the card.   The wz_mini logs don't write too much.  The temp files are typically written to `wz_mini/tmp` which is a ram disk.

An alternative would be to use something like a NAS drive or a PC and share the drive to the camera, so the camera writes everything over the network instead of the card if you are worried about the lifetime of the card.

**Comment by skilau** [comment]
I don't believe there is much worry about the write cycles to the SD card.

The camera, (assuming you have it writing to the SD), dumps about 8-10G's a day to the SD card.

The little bit of logging wz_mini_hacks does is on the order of a few K a day.
I would be shocked if this logging caused any problems at all.

PS: I purposely buy the "Endurance" cards for these cams, just so I don't have to worry so much about the write cycles.

**Comment by gtxaspec** [comment]
thinking of this, the swap file could increase load on the card too.  I wouldn't worry though, since the video writes are still the greatest threat to the card I believe.

**Comment by dudududodododedede** [comment]
Thanks guys, your input is very reassuring. This is easily my favourite open-source project

If the log files are only a few kb of new data per day then I agree it's nothing to worry about

Video writes are bad but not the worst because the writes are uniformly (continuously) distributed throughout all sectors. So the wear is even (the SD card will fail around the time when 99% of sectors are approaching their end-of-life)

I'm not sure how much writes are directed to the swap... if it's a lot and the file is small, then a small number of sectors are getting thrashed? You could end up with the SD failing because those few sectors get prematurely worn out (while 99% of sectors still have lots of useful life left)?
